src/libraries/Microsoft.Extensions.DependencyModel/src/DependencyContextLoader.cs

```diff
@@ -127,6 +127,13 @@ private DependencyContext LoadAssemblyContext(Assembly assembly, IDependencyCont
 
         private string GetDepsJsonPath(Assembly assembly)
         {
+            // Assemblies loaded in memory (e.g. single file) return empty string from Location.
+            // In these cases, don't try probing next to the assembly.
+            if (string.IsNullOrEmpty(assembly.Location))
```

Comment:
The next line will call `Assembly.Location` again. It would be nice to cache it in a local.

Reply:
It's a property, which is supposed to be fast, like a field. Does `assembly.Location` not cache it, itself?

Reply:
Also change the use later in the method to use the cached value.

Reply:
It is virtual property. It means that it will be call at least.

Reply:
Looks like the answer is "no" 😢 

https://github.com/dotnet/runtime/blob/6a80a380a1644d962137ad62bc16067bc0dbfb72/src/coreclr/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs#L458-L469

Reply:
And the current implementation allocates a new string every time.